third_party/move/move-compiler/src/parser/syntax.rs

```diff
@@ -1736,7 +1734,29 @@ fn parse_dot_or_index_chain(context: &mut Context) -> Result<Exp, Box<Diagnostic
             Tok::Period => {
                 context.tokens.advance()?;
                 let n = parse_identifier(context)?;
-                Exp_::Dot(Box::new(lhs), n)
+                let ahead = context.tokens.peek();
+                if matches!(ahead, Tok::LParen | Tok::ColonColon) {
+                    let generics = if ahead == Tok::ColonColon {
```

Comment:
I am assuming this turbofish operator `::<>` is implemented to allow for future modifications to the language? Or is there a case in the current language where we cannot type infer and need the user to explicitly specify the receiver function instantiation?